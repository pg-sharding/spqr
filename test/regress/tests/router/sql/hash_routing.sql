\c spqr-console

-- SETUP
CREATE DISTRIBUTION ds1 COLUMN TYPES varchar hash;

CREATE KEY RANGE krid1 FROM 0 ROUTE TO sh1 FOR DISTRIBUTION ds1;
CREATE KEY RANGE krid2 FROM 2147483648 ROUTE TO sh2 FOR DISTRIBUTION ds1;

-- the set of all unsigned 32-bit integers (0 to 4294967295)
ALTER DISTRIBUTION ds1 ATTACH RELATION xx DISTRIBUTION KEY col1 HASH FUNCTION MURMUR;

-- TEST
\c regress
CREATE TABLE xx (col1 varchar);
INSERT INTO xx (col1) VALUES ('Hello, world!');
INSERT INTO xx (col1) VALUES ('test');
INSERT INTO xx (col1) VALUES ('众口难调');
INSERT INTO xx (col1) VALUES ('The quick brown fox jumps over the lazy dog');
INSERT INTO xx (col1) VALUES ('Армия — не просто доброе слово, а очень быстрое дело. Так мы выигрывали все войны. Пока противник рисует карты наступления, мы меняем ландшафты, причём вручную. Когда приходит время атаки, противник теряется на незнакомой местности и приходит в полную небоеготовность. В этом смысл, в этом наша стратегия.');

--TEARDOWN
DROP TABLE xx;
\c spqr-console
DROP DISTRIBUTION ALL CASCADE;
DROP KEY RANGE ALL;