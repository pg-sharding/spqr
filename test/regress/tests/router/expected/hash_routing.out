\c spqr-console

		SPQR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

-- SETUP
CREATE DISTRIBUTION ds1 COLUMN TYPES varchar hash;
    add distribution    
------------------------
 distribution id -> ds1
(1 row)

CREATE KEY RANGE krid1 FROM 0 ROUTE TO sh1 FOR DISTRIBUTION ds1;
 add key range 
---------------
 bound -> 0
(1 row)

CREATE KEY RANGE krid2 FROM 2147483648 ROUTE TO sh2 FOR DISTRIBUTION ds1;
    add key range    
---------------------
 bound -> 2147483648
(1 row)

-- the set of all unsigned 32-bit integers (0 to 4294967295)
ALTER DISTRIBUTION ds1 ATTACH RELATION xx DISTRIBUTION KEY col1 HASH FUNCTION MURMUR;
      attach table      
------------------------
 relation name   -> xx
 distribution id -> ds1
(2 rows)

-- TEST
\c regress
CREATE TABLE xx (col1 varchar);
NOTICE: send query to shard(s) : sh1,sh2
INSERT INTO xx (col1) VALUES ('Hello, world!');
NOTICE: send query to shard(s) : sh2
INSERT INTO xx (col1) VALUES ('test');
NOTICE: send query to shard(s) : sh2
INSERT INTO xx (col1) VALUES ('众口难调');
NOTICE: send query to shard(s) : sh2
INSERT INTO xx (col1) VALUES ('The quick brown fox jumps over the lazy dog');
NOTICE: send query to shard(s) : sh1
INSERT INTO xx (col1) VALUES ('Армия — не просто доброе слово, а очень быстрое дело. Так мы выигрывали все войны. Пока противник рисует карты наступления, мы меняем ландшафты, причём вручную. Когда приходит время атаки, противник теряется на незнакомой местности и приходит в полную небоеготовность. В этом смысл, в этом наша стратегия.');
NOTICE: send query to shard(s) : sh1
--TEARDOWN
DROP TABLE xx;
NOTICE: send query to shard(s) : sh1,sh2
\c spqr-console

		SPQR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

DROP DISTRIBUTION ALL CASCADE;
   drop distribution    
------------------------
 distribution id -> ds1
(1 row)

DROP KEY RANGE ALL;
 drop key range 
----------------
(0 rows)

